src/partition/renumber.cpp

```diff
@@ -0,0 +1,65 @@
+#include "partition/renumber.hpp"
+
+#include <tbb/parallel_sort.h>
+
+namespace osrm
+{
+namespace partition
+{
+namespace
+{
+// Returns a vector that is indexed by node ID marking the level at which it is a border node
+std::vector<LevelID> getHighestBorderLevel(const DynamicEdgeBasedGraph &graph,
+                                           const std::vector<Partition> &partitions)
+{
+    std::vector<LevelID> border_level(graph.GetNumberOfNodes(), 0);
+
+    for (const auto level : util::irange<LevelID>(1, partitions.size() + 1))
+    {
+        const auto &partition = partitions[level - 1];
+        for (auto node : util::irange<NodeID>(0, graph.GetNumberOfNodes()))
+        {
+            for (auto edge : graph.GetAdjacentEdgeRange(node))
+            {
+                auto target = graph.GetTarget(edge);
+                if (partition[node] != partition[target])
+                {
+                    border_level[node] = level;
```

Comment:
but why not `border_level[node] = std::max(border_level[node], level);`?
or level is always non-decreasing -> assertion ` border_level[node] <= level;`